Library/Homebrew/upgrade.rb

```diff
@@ -249,7 +260,13 @@ def check_installed_dependents(
       quiet: false,
       verbose: false
     )
-      return if Homebrew::EnvConfig.no_installed_dependents_check?
+      if Homebrew::EnvConfig.no_installed_dependents_check?
+        opoo <<~EOS
+          HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK is set: not checking for outdated
+          dependents or dependents with broken linkage!
+        EOS
+        return
+      end
```

Comment:
I take it not being able to disable this message with `HOMEBREW_NO_ENV_HINTS` is deliberate? If so, we may wish to document that not being able to silence this is by design.

It can probably be in the description of `HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK` rather than in `HOMEBREW_NO_ENV_HINTS` (or perhaps both).

Reply:
I'd say yes because this message is prompting (albeit not explicitly) the user to go and manually upgrade their dependents when convenient.